python/test/test_basic.py

```diff
@@ -0,0 +1,64 @@
+import unittest
+from opm2.simulators import BlackOilSimulator
+import os
+
+class TestBasic(unittest.TestCase):
+    @classmethod
+    def setUpClass(cls):
+        os.chdir('test_data/SPE1CASE1')
```

Comment:
There are two problems with the `os.chdir('test_data/SPE1CASE1')`:

1. When you hardcode the `test_data/SPE1CASE` that means the code *must* be run with cwd exactly at `python/test`. If you use `test_dir = os.path.join( os.path.dirname(__file__), "test_data/SPE1CASE1")` you are guaranteed to get the correct test directory irrespective of current cwd.

2. To change cwd changes the *state* of the process - that is ugly. With Python I recommend creating a small context:
```
from contextlib import contextmanager

@contextmanager
def pushd(path):
    cwd = os.getcwd()
    if not os.path.isdir(path):
        os.makedirs(path)
    os.chdir(path)

    yield

    os.chdir(cwd)
```
Then you can in your tests temporarily change directory with:
```
with pushd("/some/path/somewhere"):
      do_stuff_in_path()
      more_stuff()
# Now we are back in the original cwd
continue_in_cwd0()
```

Reply:
Good idea. But it seems the cwd needs to stay at `...test_data/SPE1CASE1` during the whole test, so if I use `with pushd..` in `setUpClass()` the cwd is reset when that finishes, and then `step_init()` throws `Exception received: Cannot find input case 'SPE1CASE1.DATA'.`

Reply:
OK - the `pushd` should be in each of the test routines, whereas the `setUpClass` should just store the testpath.